great_expectations/rule_based_profiler/helpers/cardinality_checker.py

```diff
@@ -215,7 +220,7 @@ def _convert_to_cardinality_limit_mode(
                             f"Please specify a supported cardinality mode.  Supported cardinality modes are {[member.name for member in CardinalityLimitMode]}"
                         )
             else:
-                return cardinality_limit_mode.value
+                return cast(CardinalityLimitMode, cardinality_limit_mode).value
```

Comment:
I'm surprised to need to cast here because you handle the other cases above. If you want to avoid the cast you could do an `isinstance(cardinality_limit_mode, CardinalityLimitMode)` instead of the `else`. You could then have an `else` that raises because the type of `cardinality_limit_mode` is wrong. This is a bit more defensive, but If typechecking is happening, we should never hit this new `else` block.

Reply:
@billdirks For some reason, `PyCharm` was issuing a warning, which disappeared with the use of `cast`.

Reply:
Yeah, it probably doesn't have the logic to know you've exhausted the `Union` type. Doing an `isinstance` check is a little bit safer but type-checking should ensure the type is correct.

Reply:
This module is not being type-checked
https://github.com/great-expectations/great_expectations/blob/develop/pyproject.toml#L143